src/lib/storage/lz4_segment/lz4_encoder.hpp

```diff
@@ -168,35 +165,38 @@ class LZ4Encoder : public SegmentEncoder<LZ4Encoder> {
      * The offsets are stored as 32 bit unsigned integer as opposed to 64 bit (size_t) so that they can later be
      * compressed via vector compression.
      */
-    auto offsets = pmr_vector<uint32_t>{alloc};
-    offsets.resize(num_elements);
+    auto offsets = pmr_vector<uint32_t>{allocator};
 
     /**
      * These are the lengths of each string. They are needed to train the zstd dictionary.
      */
-    auto sample_sizes = pmr_vector<size_t>{alloc};
-    sample_sizes.resize(num_elements);
+    auto sample_sizes = pmr_vector<size_t>{allocator};
+
+    segment_iterable.with_iterators([&](auto it, auto end) {
+      const auto segment_size = std::distance(it, end);
+
+      null_values.resize(segment_size);
+      offsets.resize(segment_size);
+      sample_sizes.resize(segment_size);
```

Comment:
Legacy, but maybe `string_length_samples` is better?

Reply:
Changed to `string_samples_lengths`.